typescript/core/src/IAppendWritable.ts

```diff
@@ -0,0 +1,9 @@
+import { IWritable } from "./IWritable";
+
+/**
+ * IAppendWritable describes a writer interface with append abilities.
+ */
+export interface IAppendWritable extends IWritable {
```

Comment:
naming-wise, I'm not sure about "AppendWritable" – since being able to seek isn't necessarily related to the concept of appending. Some other ideas:

- `ISeekable`
- `ISeekableWriter`  (SeekableWritable sounds bad, but would be more consistent 😓)